src/ResourceManager/RecoveryServices.Backup/Cmdlets/Jobs/WaitAzureRmRecoveryServicesJob.cs

```diff
@@ -0,0 +1,143 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets.Models;
+using Microsoft.Azure.Commands.RecoveryServices.Backup.Helpers;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets
+{
+    [Cmdlet("Wait", "AzureRmBackupJob"), OutputType(typeof(List<AzureRmRecoveryServicesJobBase>), typeof(AzureRmRecoveryServicesJobBase))]
+    public class WaitAzureRmRecoveryServicesJob : RecoveryServicesBackupCmdletBase
+    {
+        [Parameter(Mandatory = false, HelpMessage = ParamHelpMsg.Common.Vault)]
+        public ARSVault Vault { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = ParamHelpMsg.Job.WaitJobOrListFilter)]
+        [ValidateNotNull]
+        public object Job { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = ParamHelpMsg.Job.WaitJobTimeoutFilter)]
+        public long? Timeout { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            ExecutionBlock(() =>
+            {
+                base.ExecuteCmdlet();
+
+                List<string> jobsToWaitOn = new List<string>();
+                List<AzureRmRecoveryServicesJobBase> finalJobs = new List<AzureRmRecoveryServicesJobBase>();
+
+                // TODO: Validate the following code
+                object castedObj;
+                if (GetCastedObjFromPSObj(Job, typeof(AzureRmRecoveryServicesJobBase), out castedObj))
```

Comment:
If you take JobBase as input, then you can directly use Job.GetType() and can eliminate all this casting logic